File: ios/Plugin/LottieSplashScreenPlugin.swift
import Capacitor
import Foundation

@objc(LottieSplashScreenPlugin)
public class LottieSplashScreenPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "LottieSplashScreenPlugin"
    public let jsName = "LottieSplashScreen"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "hide", returnType: CAPPluginReturnNone),
        CAPPluginMethod(name: "show", returnType: CAPPluginReturnNone),
        CAPPluginMethod(name: "appLoaded", returnType: CAPPluginReturnNone),
        CAPPluginMethod(name: "isAnimating", returnType: CAPPluginReturnPromise),
    ]
    public static var isEnabledStatic = true
    private let implementation = LottieSplashScreen()
    
    @objc func show(_ call: CAPPluginCall) {
        implementation.show();
        call.resolve();
    }
    
    @objc func hide(_ call: CAPPluginCall) {
        implementation.hide();
        call.resolve();
    }
    
    @objc func appLoaded(_ call: CAPPluginCall) {
        implementation.onAppLoaded()
        call.resolve();
    }
    
    override public func load() {
        if !LottieSplashScreenPlugin.isEnabledStatic {
            print("\(LottieSplashScreen.TAG) Not enabled statically (!?)")
            return
        }
        let isEnabled = getConfig().getBoolean("enabled", true)
        
        print("\(LottieSplashScreen.TAG) Started")
        
        if isEnabled {
            var animation = getConfig().getString("animationLight", "")
            if (animation == ""){
                NSLog("\(LottieSplashScreen.TAG) Animation must be provided in ionic.config.ts|json")
                return
            }
            var backgroundColor = getConfig().getString("backgroundLight", "#FFFFFF")
            
            let darkAnimation = getConfig().getString("animationDark", "")
            let darkBackgroundColor = getConfig().getString("backgroundDark", "#000000")
            
            let autoHide = getConfig().getBoolean("autoHide", false)
            var loopAnimation =  getConfig().getBoolean("loop", false)
            
            if autoHide, loopAnimation {
                NSLog("\(LottieSplashScreen.TAG) autoHide and loop cannot be true at the same time. Loop will be disabled.")
                loopAnimation = false
            }
            
            if #available(iOS 13.0, *), UITraitCollection.current.userInterfaceStyle == .dark {
                print("\(LottieSplashScreen.TAG) Dark mode detected. Using dark animation and color")
                if darkAnimation != "" {
                    animation = darkAnimation
                }
                if darkBackgroundColor != "" {
                    backgroundColor = darkBackgroundColor
                }
            }
            
            print("\(LottieSplashScreen.TAG) Animation path: \(animation ?? "")")
            print("\(LottieSplashScreen.TAG) Background color: \(backgroundColor ?? "")")
            print("\(LottieSplashScreen.TAG) Auto Hide: \(autoHide)")
            print("\(LottieSplashScreen.TAG) Loop Animation: \(loopAnimation)")
            
            implementation.loadLottie(
                view: self.bridge?.viewController?.view,
                path: animation,
                backgroundColor: backgroundColor,
                autoHide: autoHide,
                loopMode: loopAnimation)
        }else{
            print("\(LottieSplashScreen.TAG) Not enabled")
        }
        implementation.onAnimationEvent = onAnimationEvent
    }
    
    public func onAnimationEvent(event: AnimationEventListener) {
        print("\(LottieSplashScreen.TAG) onAnimationEvent", event.listenerEvent)
        self.bridge?.triggerWindowJSEvent(eventName: event.listenerEvent)
        self.notifyListeners(event.listenerEvent, data: nil)
    }
    
    @objc func isAnimating(_ call: CAPPluginCall) {
        call.resolve([
            "isAnimating": implementation.isAnimating()
        ])
    }
    
}

File: android/src/main/java/com/ludufre/plugins/lottiesplashscreen/LottieSplashScreenPlugin.java
package com.ludufre.plugins.lottiesplashscreen;

import android.content.Context;
import android.content.res.Configuration;
import android.os.Build;
import android.util.Log;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

import java.util.Objects;

@CapacitorPlugin(name = "LottieSplashScreen")
public class LottieSplashScreenPlugin extends Plugin {
    private final LottieSplashScreen implementation = new LottieSplashScreen();
    public static boolean isEnabledStatic = true;

    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
    public void show(PluginCall call) {
        ShowLottieSplashScreenDialog();
        call.resolve();
    }

    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
    public void hide(PluginCall call) {
        implementation.hideDialog();
        call.resolve();
    }

    @PluginMethod(returnType = PluginMethod.RETURN_NONE)
    public void appLoaded(PluginCall call) {
        implementation.onAppLoaded();
        call.resolve();
    }

    @PluginMethod
    public void isAnimating(PluginCall call) {
        boolean isAnimating = implementation.isAnimating();
        JSObject ret = new JSObject();
        ret.put("isAnimating", isAnimating);
        call.resolve(ret);
    }

    public void load() {
        if (!isEnabledStatic) {
            Log.e(LottieSplashScreen.TAG, "Not enabled statically (!?)");
            return;
        }
        boolean isEnabled = this.getConfig().getBoolean("enabled", true);

        Log.i(LottieSplashScreen.TAG, "Started");

        if (isEnabled) {
            ShowLottieSplashScreenDialog();
        } else {
            Log.i(LottieSplashScreen.TAG, "Not enabled");
        }
        implementation.setAnimationEventListener(this::onAnimationEvent);
    }

    public void onAnimationEvent(String event) {
        Log.i(LottieSplashScreen.TAG, "onAnimationEvent");
        bridge.triggerWindowJSEvent(event);
        notifyListeners(event, null);
    }

    private void ShowLottieSplashScreenDialog() {
        Context context = this.getContext();

        String animation = this.getConfig().getString("animationLight", "");
        if (animation == "") {
            Log.e(LottieSplashScreen.TAG, "Animation must be provided in ionic.config.ts|json");
            return;
        }
        String backgroundColor = this.getConfig().getString("backgroundLight", "#FFFFFF");

        String darkAnimation = getConfig().getString("animationDark", "");
        String darkBackgroundcolor = getConfig().getString("backgroundDark", "#000000");

        boolean autoHide = getConfig().getBoolean("autoHide", false);
        boolean loopAnimation = getConfig().getBoolean("loop", false);

        if (autoHide && loopAnimation) {
            Log.e(LottieSplashScreen.TAG, "autoHide and loop cannot be true at the same time. Loop will be disabled.");
            loopAnimation = false;
        }

        if (isDarkMode()) {
            Log.i(LottieSplashScreen.TAG, "Dark mode detected. Using dark animation and color");
            if (!Objects.equals(darkAnimation, "")) {
                animation = darkAnimation;
            }
            if (!Objects.equals(darkBackgroundcolor, "")) {
                backgroundColor = darkBackgroundcolor;
            }
        }

        Log.i(LottieSplashScreen.TAG, "Animation path: " + animation);
        Log.i(LottieSplashScreen.TAG, "Background color: " + backgroundColor);
        Log.i(LottieSplashScreen.TAG, "Auto Hide: " + autoHide);
        Log.i(LottieSplashScreen.TAG, "Loop Animation: " + loopAnimation);

        implementation.ShowLottieSplashScreenDialog(context, animation, backgroundColor, autoHide, loopAnimation);
    }

    private boolean isDarkMode(){
        if(Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
        return false;
        }

        int nightModeFlags =
                getContext().getResources().getConfiguration().uiMode &
                        Configuration.UI_MODE_NIGHT_MASK;
        return switch (nightModeFlags) {
            case Configuration.UI_MODE_NIGHT_YES -> true;
            case Configuration.UI_MODE_NIGHT_NO -> false;
            case Configuration.UI_MODE_NIGHT_UNDEFINED -> false;
            default -> false;
        };
    }
}
